compiler/rustc_interface/src/callbacks.rs

```diff
@@ -25,6 +25,16 @@ fn span_debug(span: rustc_span::Span, f: &mut fmt::Formatter<'_>) -> fmt::Result
     })
 }
 
+fn track_span_parent(def_id: rustc_span::def_id::LocalDefId) {
+    tls::with_opt(|tcx| {
```

Comment:
This callback looks like it introduces overhead even if `-Z incremental_relative_spans` is disabled. Maybe that's the source of the regression?

Reply:
Nevermind, I just saw that the callback only gets invoked if parent is `Some(...)` which won't be the case if `-Z incremental_relative_spans` is disabled.